volatility3/framework/plugins/windows/callbacks.py

```diff
@@ -65,10 +208,133 @@ def create_callback_table(
             config_path,
             "windows",
             symbol_filename,
+            class_types=callbacks.class_types,
             native_types=native_types,
             table_mapping=table_mapping,
         )
 
+    @classmethod
+    def scan(
+        cls,
+        context: interfaces.context.ContextInterface,
+        layer_name: str,
+        nt_symbol_table: str,
+        callback_symbol_table: str,
+    ) -> Iterable[
+        Tuple[
+            Union[str, interfaces.renderers.BaseAbsentValue],
+            int,
+            Union[str, interfaces.renderers.BaseAbsentValue],
+        ]
+    ]:
+        """Scans for callback objects using the poolscanner module and constraints.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The name of the layer on which to operate
+            nt_symbol_table: The name of the table containing the kernel symbols
+            callback_symbol_table: The name of the table containing the callback object symbols (_SHUTDOWN_PACKET etc.)
+
+        Returns:
+            A list of callback objects found by scanning the `layer_name` layer for callback pool signatures
+        """
+        is_vista_or_later = versions.is_vista_or_later(
+            context=context, symbol_table=nt_symbol_table
+        )
+
+        type_map = handles.Handles.get_type_map(context, layer_name, nt_symbol_table)
+
+        constraints = cls.create_callback_scan_constraints(
+            context, callback_symbol_table, is_vista_or_later
+        )
+
+        for (
+            _constraint,
+            mem_object,
+            _header,
+        ) in poolscanner.PoolScanner.generate_pool_scan(
+            context, layer_name, nt_symbol_table, constraints
+        ):
+            try:
+                if hasattr(mem_object, "is_valid") and not mem_object.is_valid():
```

Comment:
The check fails for `StructType` objects if we remove the `hasattr` check. For instance: `AttributeError: StructType has no attribute: callbacks-x641!_NOTIFICATION_PACKET.is_valid`

There is an explicitly defined `is_valid` method in the callbacks extension added as part of this PR, but I don't think any other objects will have an `is_valid` method since they are just `StructType` objects defined in the callbacks symbol files.